### PR TITLE
Block copy stats

### PR DESCRIPTION
**Pull request by mheap**
## Description

Adds a copy to clipboard button to building blocks. It works, but needs some design love.

screen shot 2018-08-27 at 15 09 31

Clicking copy will copy dependency configuration *and* the code itself as people seem to miss the dependency config section.

This only works on blocks using the `building_block` filter (currently Voice)

When the user clicks on the copy button for the first time per page load, we store which language they have selected, and which building block they're using.

## Deploy Notes

This adds a new DB migration to create the usage_copy_to_clipboard table. I believe they're run automatically but we should take care when deploying this

### CODE REVIEW

**Comment by mheap** [comment]
I've added more visual clues to the icon as I don't think it was obvious what the button did

screen shot 2018-08-27 at 15 18 02

Once you click:

screen shot 2018-08-27 at 15 18 08

File: db/migrate/20180827133110_usage_copy_to_clipboard.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class UsageCopyToClipboard < ActiveRecord::Migration[5.1]
```
Comment (cr0wst) [inline db/migrate/20180827133110_usage_copy_to_clipboard.rb]: In `devrel-private` we talked about the possibility of making this table less granular so we can use it for other event tracking.

**Comment by cr0wst** [comment]
Just a few discussion items from me. I like it, though. 👍

**Comment by mheap** [comment]
@cr0wst I refactored to a more generic event store and now track links on dependency and main code source links in addition to copy to clipboard

**Comment by lornajane** [comment]
I'm not sure about copying the dependencies. It sort of makes sense in PHP but makes no sense in Java where the two blocks don't belong next to one another and once they're in your clipboard, that's much less obvious. As a user, I would expect the "copy" link to only copy the block it was attached to.

**Comment by mheap** [comment]
@lornajane I think it makes sense in Java too. `Initialize your dependencies` contains things like the following that are incidental, but required:

```
AuthMethod auth = new JWTAuthMethod(NEXMO_APPLICATION_ID, FileSystems.getDefault().getPath(NEXMO_PRIVATE_KEY));
NexmoClient nexmo = new NexmoClient(auth);
```

https://developer.nexmo.com/voice/voice-api/building-blocks/make-an-outbound-call/java

For me, I'd expect to have that copied

**Comment by lornajane** [comment]
Right but these are two non-continuous blocks of code that might go in different locations or might need other stuff in between (my Java is rusty and my .NET non-existent, I'm judging this on what I see when I click the "view full source code" link).  We're offering to copy something that a) isn't what is in the box with the copy button attached and b) isn't a complete example.  I think it's less confusing to have the copy button copy what's in the box that the button is attached to since we can't really solve the problem of providing all the context that might be required to go with it.

**Comment by mheap** [comment]
I don't feel strongly about copying both, so let's scope the copy to just the block selected and tackle the dependency configuration issue in another PR. I'll make the relevant changes

**Comment by mheap** [comment]
@lornajane I've updated the branch with what we discussed. Could you review and make sure it does what you expect?

**Pull request merged**